Cannot run npm install with custom modules with Nuxt 3.5.0

### Environment

```
------------------------------
- Operating System: Linux
- Node Version:     v16.14.2
- Nuxt Version:     3.5.0
- Nitro Version:    2.4.0
- Package Manager:  npm@7.17.0
- Builder:          vite
- User Config:      modules
- Runtime Modules:  ./modules/myModule
- Build Modules:    -
------------------------------
```

### Reproduction

Run `npm install` in this sandbox: 
https://stackblitz.com/edit/github-s6insh?file=nuxt.config.ts

### Describe the bug

Running `npm install` with a custom module results in an error that the module cannot be found. I wasn't expecting an error. This previously worked fine with Nuxt 3.4.3.

```
ERROR  Error while requiring module ./modules/myModule: Error: Cannot find module 'modules/myModule'           5:15:13 PM
Require stack:
- /home/projects/github-s6insh/index.js


 ERROR  Cannot start nuxt:  Cannot find module 'modules/myModule'                                               5:15:13 PM
Require stack:                                                                                                  5:15:07 PM
- /home/projects/github-s6insh/index.js

  Require stack:
  - index.js
  at Module._resolveFilename (https://githubs6insh-mmv3.w-corp.staticblitz.com/blitz.bdcb25d8.js:35:242305)
  at Function.resolve (https://githubs6insh-mmv3.w-corp.staticblitz.com/blitz.bdcb25d8.js:35:480406)
  at Function._resolve [as resolve] (node_modules/jiti/dist/jiti.js:1:241100)
  at resolveModule (file://node_modules/@nuxt/kit/dist/index.mjs:331:36)
  at requireModule (file://node_modules/@nuxt/kit/dist/index.mjs:341:24)
  at normalizeModule (file://node_modules/@nuxt/kit/dist/index.mjs:537:90)
  at async Module.installModule (file://node_modules/@nuxt/kit/dist/index.mjs:510:41)
  at async initNuxt (file://node_modules/nuxt/dist/index.mjs:2906:7)
  at async load (file://node_modules/nuxi/dist/chunks/dev.mjs:216:9)
  at async Object.invoke (file://node_modules/nuxi/dist/chunks/dev.mjs:260:5)
```

### Additional context

_No response_

### Logs

_No response_

That also results in an error:
```
ERROR  Cannot find module 'modules/myModule/index'
Require stack:
- /home/projects/github-s6insh-eawouj/index.js
```

index.js indeed doesn't exist, it should be index.ts.

--- a/packages/kit/src/module/install.ts
+++ b/packages/kit/src/module/install.ts
@@ -1,6 +1,6 @@
 import { lstatSync } from 'node:fs'
 import type { Nuxt, NuxtModule } from '@nuxt/schema'
-import { dirname, isAbsolute, normalize } from 'pathe'
+import { dirname, isAbsolute, normalize, resolve } from 'pathe'
 import { isNuxt2 } from '../compatibility'
 import { useNuxt } from '../context'
 import { requireModule } from '../internal/cjs'
@@ -53,7 +53,11 @@ async function normalizeModule (nuxtModule: string | NuxtModule, inlineOptions?:
 
   // Import if input is string
   if (typeof nuxtModule === 'string') {
-    const src = normalize(resolveAlias(nuxtModule))
+    let src = resolveAlias(nuxtModule)
+    if (src.match(/^\.{1,2}\//)) {
+      src = resolve(nuxt.options.rootDir, src)
+    }
+    src = normalize(src)
     try {
       // Prefer ESM resolution if possible
       nuxtModule = await importModule(src, nuxt.options.modulesDir).catch(() => null) ?? requireModule(src, { paths: nuxt.options.modulesDir })

--- a/test/fixtures/basic/modules/test/index.ts
+++ b/test/fixtures/basic/modules/test/index.ts
@@ -1,0 +1,7 @@
+import { defineNuxtModule } from 'nuxt/kit'
+
+export default defineNuxtModule({
+  meta: {
+    name: 'test'
+  }
+})

--- a/test/fixtures/basic/nuxt.config.ts
+++ b/test/fixtures/basic/nuxt.config.ts
@@ -80,6 +80,7 @@ export default defineNuxtConfig({
     }
   },
   modules: [
+    './modules/test/index',
     [
       '~/modules/example',
       {

